### PR TITLE
SDK-Core Release Subgraph Compatibility Tests

### PR DESCRIPTION
**Pull request by 0xdavinchee**
## General Refactoring
- General CI/CD file cleanup (quotes, unused env, etc.)
- `hardhat.config.ts` chainId of localhost set to hardhat node chainId and comment cleanup
- deprecate usage of ganache, use hardhat node instead
- split startGanacheAndDeployContracts.sh => startHardhatNode.sh and setupTestEnvironment.sh

## CI/CD Updates
- `generate-graphql-schema` run before build and release/testing in ci.feature/ci.canary/ci.release + use env. SUBGRAPH_RELEASE_TAG: RELEASE_VERSION to properly test each respective endpoint in the different CI builds
- new CI file which runs tests on PRs to `release-subgraph-v1` or `release-sdk-core-stable`
  - deploy local subgraph + sdk-core test (done on PRs to either branch)
  - build and test live subgraph with current sdk-core release (done only on PRs to `release-sdk-core-stable` branch)
  - build and test previous sdk-core versions with current subgraph release (done only on PRs to `release-subgraph-v1` branch)
  - refactored CI with the use of reusable workflows:
     - `call.check-subgraph-indexing-statuses`: ensure subgraph fully indexed to all networks before SDK-core can be released
     - `setup-deploy-and-test-local-subgraph`: deploy local subgraph and run subgraph package tests: feature/canary or sdk-core tests: release branch
     - `test-sdk-core`: install, build and test SDK-core - takes subgraph-release as input (feature/dev/v1)
     - `test-subgraph-on-previous-sdk-core-versions`: runs query tests on either TBD subgraph or live and deployed subgraphs on previous versions of sdk-core (done for dev and release CI)
> NOTE: `uses:  superfluid-finance/protocol-monorepo/.github/workflows/call.<NAME>.yml@<BRANCH | COMMIT>` specifies a commit/branch of the reusable workflow to use whereas `uses: ./.github/workflows/` uses commit which triggered the workflow.

### CODE REVIEW

File: .github/workflows/call.setup-deploy-and-test-local-subgraph.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
+name: Set up, deploy and test local subgraph
```
Comment (hellwolf) [inline .github/workflows/call.setup-deploy-and-test-local-subgraph.yml]: Maybe we should be consistent about applying https://erinwrightwriting.com/what-is-headline-style-capitalization/ to name fields...

File: .github/workflows/ci.canary.yml [hunk 2]
```diff
@@ -54,52 +54,17 @@ jobs:
         env:
           POLYGON_MAINNET_PROVIDER_URL: ${{ secrets.POLYGON_MAINNET_PROVIDER_URL }}
 
-  subgraph-test:
-    name: Run test of subgraph on dev branch
-
+  test-subgraph:
+    uses: superfluid-finance/protocol-monorepo/.github/workflows/call.setup-deploy-and-test-local-subgraph.yml@dev
```
Comment (hellwolf) [inline .github/workflows/ci.canary.yml:60]: I would add these dependent "call workflow" yml files to the "paths" dependents list.

File: .github/workflows/ci.canary.yml [hunk 3]
```diff
@@ -54,52 +54,17 @@ jobs:
...
-          ./setup.sh
-        working-directory: ./graph-node/docker
-
-      - name: "Docker compose"
-        run: docker-compose up &
-        working-directory: ./graph-node/docker
-
-      - name: "Run integration test suite"
-        run: yarn workspace @superfluid-finance/subgraph test
+  test-subgraph-on-previous-sdk-versions:
```
Comment (hellwolf) [inline .github/workflows/ci.canary.yml:64]: same:



I would add these dependent "call workflow" yml files to the "paths" dependents list.

File: .github/workflows/ci.feature.yml [hunk 4]
```diff
@@ -140,100 +140,18 @@ jobs:
...
-    name: Build and test subgraph (Feature Branch)
-
-    runs-on: ubuntu-latest
-
-    env:
-      subgraph-working-directory: ./packages/subgraph
-      contracts-working-directory: ./packages/ethereum-contracts
-      js-sdk-working-directory: ./packages/js-sdk
-
+    uses: ./.github/workflows/call.setup-deploy-and-test-local-subgraph.yml
```
Comment (hellwolf) [inline .github/workflows/ci.feature.yml:146]: ditto

File: .github/workflows/handler.publish-release-packages.yml [hunk 5]
```diff
@@ -42,6 +42,7 @@ jobs:
       - name: Build package
         run: |
           yarn install --frozen-lockfile
+          yarn --cwd packages/sdk-core generate-graphql-schema:v1
```
Comment (hellwolf) [inline .github/workflows/handler.publish-release-packages.yml:45]: okay

File: .github/workflows/ci.feature.yml [hunk 6]
```diff
@@ -140,100 +140,18 @@ jobs:
...
-      - name: "Build contracts"
-        run: yarn run build:contracts
-        working-directory: ${{ env.contracts-working-directory }}
-
-      - name: "Run test suite"
-        run:
-          yarn test
-        working-directory: ${{ env.sdk-core-working-directory }}
+    with:
+      subgraph-release: feature
```
Comment (hellwolf) [inline .github/workflows/ci.feature.yml:157]: 🆒

File: .github/workflows/ci.canary.yml [hunk 7]
```diff
@@ -54,52 +54,17 @@ jobs:
...
+    uses: superfluid-finance/protocol-monorepo/.github/workflows/call.test-subgraph-on-previous-sdk-core-versions.yml@dev
+    name: Build and Test Subgraph on Previous SDK-Core Versions (Development Branch)
+    if: github.repository == 'superfluid-finance/protocol-monorepo'
+    with:
+      subgraph-release: dev
```
Comment (hellwolf) [inline .github/workflows/ci.canary.yml:69]: 🆒

File: .github/workflows/ci.pre-release-draft-test-sdk-core-subgraph-integration.yml [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+      local-subgraph-url: http://localhost:8000/subgraphs/name/superfluid-test
+      subgraph-release: v1
+
+  # ensure subgraph indexing is complete on all v1 endpoints before allowing sdk-core release
+  check-subgraph-indexing-completeness:
+    uses: ./.github/workflows/call.check-subgraph-indexing-statuses.yml
+    name: Check Subgraph Indexing Status for v1 Endpoints
+    if: github.base_ref == 'release-sdk-core-stable'
+    with:
+      subgraph-release: v1
```
Comment (hellwolf) [inline .github/workflows/ci.pre-release-draft-test-sdk-core-subgraph-integration.yml:47]: 🆒

File: .github/workflows/ci.pre-release-draft-test-sdk-core-subgraph-integration.yml [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+  # build and test live subgraph with release sdk-core
+  build-and-test-live-subgraph-current-release:
+    uses: ./.github/workflows/call.test-sdk-core.yml
+    name: Build and Test SDK-Core (Release branch)
+    if: github.base_ref == 'release-sdk-core-stable'
+    with:
+      subgraph-release: v1
```
Comment (hellwolf) [inline .github/workflows/ci.pre-release-draft-test-sdk-core-subgraph-integration.yml:55]: 🆒

**Review by hellwolf — APPROVED** [review]
good stuff, added some minor requests.

**Comment by 0xdavinchee** [comment]
@hellwolf addressed the changes + prepended name of workflows with:
- Reusable Workflow
- CI |
- CD |
- nothing for handler workflows

**Pull request merged**